snuba_sdk/delete_query.py

```diff
@@ -0,0 +1,46 @@
+import json
+from dataclasses import dataclass
+from typing import Any, Dict, List, Union
+
+from snuba_sdk.query import BaseQuery
+
+
+@dataclass(frozen=True)
+class DeleteQuery(BaseQuery):
+    """
+    This represents a snuba delete query.
+    Inputs:
+        storage - the storage to delete from
+        columnConditions - a mapping from column-name to a list of column values
+            that defines the delete conditions. ex:
+            {
+                "id": [1, 2, 3]
+                "status": ["failed"]
+            }
+            represents
+            DELETE FROM ... WHERE id in (1,2,3) AND status='failed'
+    Deletes all rows in the given storage, that satisfy the conditions
+    defined in 'columnConditions'.
+    """
+
+    storage_name: str
+    column_conditions: Dict[str, List[Any]]
+
+    def validate(self) -> None:
+        """
+        i dont think we need to do any input validation at the sdk
+        level bc inputs will be validated at the endpoint and the proper
+        response code returned.
+        """
+        return
+
+    def serialize(self) -> Union[str, Dict[str, Any]]:
+        # the body of the request
+        self.validate()
+        return json.dumps({"columns": self.column_conditions})
```

Comment:
storage is in the url not the body, which is why its not in the serialize. it will be grabbed off the object later by `raw_snql_query` to put into the url.